### PR TITLE
Split janestreet camlp4 extensions

### PR DESCRIPTION
**Pull request by diml**
The next release of janestreet packages will  be camlp4-free.  For packages that currently contain both a library and a camlp4 syntax extension, this PR adds a new camlp4-only package:

package | camlp4-only package
-----------|-----------------------------
sexplib   | pa\_sexp\_conv
bin_prot | pa\_bin\_prot
typerep | pa\_typerep\_conv
fieldslib | pa\_fields\_conv
variantslib | pa\_variants\_conv

For every package in opam that depends on both on of this package and type\_conv, this PR also add a constraint so that it will be incompatible with the next jane street release

### CODE REVIEW

**Comment by avsm** [comment]
Thanks! One annoyance is that this will break upstream `opam` files in repositories. I wonder if it's worth starting a global `CHANGES.md` file that records significant changes like this that will apply to a broad set of packages.  Thoughts?

**Comment by diml** [comment]
Yeah, a global changelog seems good. We could also automatically notify maintainers when a patch touching one of their package is committed to the opam-repository

**Comment by avsm** [comment]
Let's start it off with this changeset then, and I'll add more as they come in.  How about a `CHANGES.md` file in the root of the repository?

**Comment by dsheets** [comment]
#4897 is related.

**Comment by diml** [comment]
I added a CHANGES.md with upgrade instructions

File: CHANGES.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+## Camlp4 syntax extensions split from janestreet packages (2016-01-11)
+
+Jane street packages no longer support camlp4 after the 113.09.00
```
Comment (Leonidas-from-XIV) [inline CHANGES.md]: "Jane Street"

File: CHANGES.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+series.
+
+For Jane Street packages that contain both a library and a camlp4
+syntax extension, the latter is extracted to its own package.
+
+The following table summarize the changes:
+
+Package     | Old finlib name of syntax | New camlp4-only package | version
```
Comment (Leonidas-from-XIV) [inline CHANGES.md]: "finlib" -> "findlib"

**Comment by diml** [comment]
@Leonidas-from-XIV, thanks for reviewing, I fixed the typos

**Comment by Leonidas-from-XIV** [comment]
@diml Great, looking forward to the release!

**Comment by samoht** [comment]
Seems that ackdo is broken and so limit the testability of this PR. Maybe we should mark that package as deprecated somehow?

```
#=== ERROR while installing ackdo-reloaded.0.3 ================================#
# opam-version 1.2.2
# os           linux
# command      make opaminstall BINDIR=/home/travis/.opam/system/bin
# path         /home/travis/.opam/system/build/ackdo-reloaded.0.3
# compiler     system (4.02.3)
# exit-code    2
# env-file     /home/travis/.opam/system/build/ackdo-reloaded.0.3/ackdo-reloaded-7400-0ac68e.env
# stdout-file  /home/travis/.opam/system/build/ackdo-reloaded.0.3/ackdo-reloaded-7400-0ac68e.out
# stderr-file  /home/travis/.opam/system/build/ackdo-reloaded.0.3/ackdo-reloaded-7400-0ac68e.err
### stdout ###
# cp ackdo.native /home/travis/.opam/system/bin/ackdo
### stderr ###
# cp: cannot stat `ackdo.native': No such file or directory
# make: *** [opaminstall] Error 1
```

**Comment by rgrinberg** [comment]
@samoht I'm the author and I'm fine with just removing it.

**Comment by diml** [comment]
@rgrinberg, BTW I had a quick look at the Makefile of ackdo-reloaded, I think the problem is that it is trying to copy the symlink created by ocamlbuild. The install rule should be instead:

```Makefile
install:
	cp _build/src/ackdo.native $(bin)/ackdo
```

same for the opaminstall rule

**Comment by samoht** [comment]
@rgrinberg can you submit a PR to fix/remove it? Thanks!

**Comment by samoht** [comment]
@diml can you rebase to master? ackdo-reloaded has been removed in #5471

**Comment by diml** [comment]
done

**Comment by diml** [comment]
The build is now failing because of zmq; I think it's a problem with the version of the C library. I can install zmq fine with libzmq 4.0.5 and according to the travis log it's using libzmq 3.2.2

**Comment by diml** [comment]
Same problem as #5413 it seems. @avsm, any update on Docker-based container support for Travis?

**Comment by avsm** [comment]
manually tested, will deal with the Travis/Docker writeup shortly as a separate PR.

**Comment by diml** [comment]
Thanks!

**Pull request merged**